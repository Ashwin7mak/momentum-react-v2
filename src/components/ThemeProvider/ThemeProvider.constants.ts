const CLASSES = {
  root: 'md-theme-provider',
};

const THEME_CLASS_PREFIX = 'md-theme';

export {
  CLASSES,
  THEME_CLASS_PREFIX,
};
