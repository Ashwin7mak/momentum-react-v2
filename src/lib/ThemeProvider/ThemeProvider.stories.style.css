.md-theme-storybook-example-theme {
  --sbe-bg-color: #333;
  --sbe-font-color: #eee;
}
