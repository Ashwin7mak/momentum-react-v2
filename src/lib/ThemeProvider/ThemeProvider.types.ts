import { ReactNode } from 'react';

export interface ThemeProviderProps {
  children?: ReactNode;
  token?: string;
}
