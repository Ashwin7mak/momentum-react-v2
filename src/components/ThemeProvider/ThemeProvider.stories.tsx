import React from 'react';
import { Story } from '@storybook/react';

import ThemeProvider, {
  ThemeProviderProps as Props,
  THEME_PROVIDER_CONSTANTS as CONSTANTS
} from './';

export default {
  title: 'Momentum UI/ThemeProvider',
  component: ThemeProvider,
  argTypes: {
    theme: {
      defaultValue: CONSTANTS.DEFAULTS.THEME,
      description: 'The name of the target theme to apply to all child nodes of this `<ThemeProvider />`.',
      options: [...Object.values(CONSTANTS.THEME_NAMES)],
      control: { type: 'select' },
      table: {
        type: {
          sumary: 'string'
        },
        defaultValue: {
          summary: CONSTANTS.DEFAULTS.THEME,
        },
      },
    },
    children: {
      defaultValue: 'Example Text',
      description: 'Provides the child nodes for this element. The child nodes of this element are themed by the assigned theme.',
      control: { type: 'text' },
      table: {
        type: {
          summary: 'ReactNode',
        },
        defaultValue: {
          summary: 'undefined',
        },
      },
    },
  },
};

const Template: Story<Props> = (args) => {
  const { children } = args;
  const mutatedArgs = { ...args };

  return (
    <ThemeProvider {...mutatedArgs}>
      <div style={{
        backgroundColor: 'var(--theme-background-solid-normal)',
        padding: '2rem',
      }}>
        <div style={{
          alignItems: 'center',
          backgroundColor: 'var(--button-primary-background, #eee)',
          color: 'var(--button-primary-text, #333)',
          display: 'flex',
          height: '4rem',
          justifyContent: 'center',
          width: '16rem',
        }}>{children}</div>
      </div>
    </ThemeProvider>
  );
};

const Example = Template.bind({});

Example.args = {};

export {
  Example,
};
