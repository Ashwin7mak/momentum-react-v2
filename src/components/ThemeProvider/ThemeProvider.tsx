import React, { FC } from 'react';

import { CLASSES, THEME_CLASS_PREFIX } from './ThemeProvider.constants';
import { Props } from './ThemeProvider.types';
import './ThemeProvider.style.css';

/**
 * Provides a collection of CSSVariables based on a ThemeToken to all child elements.
 */
const ThemeProvider: FC<Props> = ({ children, theme }: Props) => (
  <div className={`${CLASSES.root} ${THEME_CLASS_PREFIX}-${theme}`}>{children}</div>
);

export default ThemeProvider;
