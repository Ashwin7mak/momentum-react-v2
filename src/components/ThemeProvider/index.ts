import { default as ThemeProvider } from './ThemeProvider';
import { Props } from './ThemeProvider.types';

export type ThemeProviderProps = Props;

export default ThemeProvider;
