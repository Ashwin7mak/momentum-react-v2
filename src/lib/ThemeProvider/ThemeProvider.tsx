import React, { FC } from 'react';

import { CLASSES, THEME_CLASS_PREFIX } from './ThemeProvider.constants';
import { ThemeProviderProps as Props } from './ThemeProvider.types';
import './ThemeProvider.style.css';

/**
 * Provides a collection of CSSVariables based on a ThemeToken to all child elements.
 */
const ThemeProvider: FC<Props> = ({ children, token }: Props) => (
  <div className={`${CLASSES.main} ${THEME_CLASS_PREFIX}-${token}`}>{children}</div>
);

export default ThemeProvider;