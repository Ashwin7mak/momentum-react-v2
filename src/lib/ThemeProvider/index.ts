import { default as ThemeProvider } from './ThemeProvider';
import { ThemeProviderProps as Props } from './ThemeProvider.types';

export type ThemeProviderProps = Props;

export default ThemeProvider;
