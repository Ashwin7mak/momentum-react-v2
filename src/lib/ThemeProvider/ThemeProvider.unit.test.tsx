import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from '@momentum-ui/react';

import { CLASSES, THEME_CLASS_PREFIX } from './ThemeProvider.constants';

describe('<ThemeProvider />', () => {
  let container;

  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      container = mount(<ThemeProvider token="test" />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its main class', () => {
      expect.assertions(1);

      const component = mount(<ThemeProvider />).childAt(0);

      expect(component.hasClass(CLASSES.main)).toBe(true);
    });

    it('should have an abstracted token class', () => {
      expect.assertions(1);

      const token = 'test';

      const component = mount(<ThemeProvider token={token} />).childAt(0);

      expect(component.hasClass(`${THEME_CLASS_PREFIX}-${token}`)).toBe(true);
    });

    it('should mount children to the component nodes', () => {
      expect.assertions(1);

      const children = [
        <div key="0" />,
        <div key="1" />,
        <div key="2" />,
      ];

      const component = mount(<ThemeProvider>{children}</ThemeProvider>).childAt(0);

      expect(component.children().length).toBe(children.length);
    });
  });
});
