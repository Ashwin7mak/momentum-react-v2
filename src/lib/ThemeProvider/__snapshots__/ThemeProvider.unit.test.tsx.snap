// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeProvider /> snapshot should match snapshot 1`] = `
<ThemeProvider
  token="test"
>
  <div
    className="md-theme-provider md-theme-test"
  />
</ThemeProvider>
`;
